spec/util/extensions/miq-module_spec.rb

```diff
@@ -41,6 +41,22 @@
       expect(test_module.default).to eq(value)
     end
 
+    it 'will return a prior cached value if resetting the value fails' do
+      $already_run = false
+      test_class.cache_with_timeout(:flapping_method) do
+        if $already_run == false
+          $already_run = true
+          5
+        else
+          raise "OOPS"
```

Comment:
> It makes no sense to clear the cache if we can't store a new value in the cache. We'd cache a nil.

No, the cache would not exist (it would be no different that the value before the very first time calling the method).